### DISCUSSION TITLE
npm workspace missing module on install

### DISCUSSION THREAD

**Discussion by withinsight** [community/community#123972] [npm]
I'm using npm workspaces for the first time, and I'm encountering something that feels like a bug. I have a single root `package.json` which lists my 5 packages (we'll call them `pkg1`, `pkg2`, etc.). Here's the `dependency` setup between packages:

`pkg1`: none
`pkg2`: none
`pkg3`: `pkg1`, `pkg2`
`pkg4`: `pkg1`, `pkg2`, `pkg3`
`pkg5`: `pkg1`, `pkg2`, `pkg3`

When I run `npm i` in the monorepo root, and review the `node_modules` folders for each package, everything installs properly, _except for one dependency_: the `pkg3` dependency in `pkg5`. 

It doesn't feel like a hoisting issue, as in my scoped folder in the root `node_modules`, all that exists are symlinks, which would indicate that `pkg5` isn't getting `pkg3` from a `node_modules` further up as a phantom dependency.

I tried running `npm install` with the `install-strategy` set to `nested` and `linked`, and nothing changes (indicating its not hoisting). I'm stuck because I can't figure out why npm would install `pkg3` properly in `pkg4`, but not `pkg5`.

**Comment by withinsight** [comment]
I figured this out; it was a version mismatch. We had:

`pkg4`: `pkg3` and
`pkg5`: `^pkg3`

which, can result in a version mismatch. A couple things: 

1. the version of `pkg3` available in the repository was the same for both, so should _not have_ caused a version mismatch
2. it silently fails when installing

So rather than notifying users of packages that aren't installed when using npm workspaces due to different versions being installed in different packages is one layer of obfuscation. The second is that the versions weren't actually mismatched, because the one specified in `pkg4` and the one speciffied in `pkg5` should have resolved to the same version given what was available. 

It would be great if npm logged errors when packages aren't installed, with helpful messaging. Would have saved me a day or two (and I'm sure others many, many more).

**Comment by grandmas-favorite** [comment]
Hi @withinsight!

Thanks so much for returning to the Community to share your solution! This will certainly help others who come across the same issue.

 I agree that extra logging would be useful, thanks for the product feedback as well!
